### PR TITLE
Added static File methods for directory creation

### LINKED ISSUE #1701: Add static createDirectory + createDirectories methods to File
And use them in unit tests.

### PR DESCRIPTION
**Pull request by marcusnaslund**
Fixes #1701

### CODE REVIEW

File: source/system/io/File.ooc [hunk 1]
```diff
@@ -278,15 +278,27 @@ File: abstract class {
 	rename: static func (originalPath, newPath: String) {
 		rename(originalPath toCString(), newPath toCString())
 	}
-	getCwd: static func -> String {
-		ooc_get_cwd()
+	createDirectory: static func ~file (path: String) -> Int {
```
Comment (sebastianbaginski) [inline source/system/io/File.ooc]: @marcusnaslund  do we even need this one ?

Comment (marcusnaslund) [inline source/system/io/File.ooc]: Not really, but it is the helper function that `createDirectories` uses, and it's different for the unix and win32 implementations. We can probably make it better though, but not now.

Comment (marcusnaslund) [inline source/system/io/File.ooc]: Oh, sorry, now I see where you commented. You're right, we can remove this one.

**Comment by marcusnaslund** [comment]
Since you're already here, peer review @sebastianbaginski ?

**Comment by sebastianbaginski** [comment]
Looks good.
In the future I guess it could be convenient to have another `Dir` class which could handle all the stuff related to directory access (instead of having this inside `File`). I mean, I think it's better to have a `Dir` object with specialized interface (like listing entries, iterating etc.) than a `File` object with `file isDirectory == true`.

**Comment by sebastianbaginski** [comment]
Sorry I can't see what's wrong with PRInt from here.

File: source/system/io/File.ooc [hunk 2]
```diff
@@ -278,15 +278,28 @@ File: abstract class {
...
+	createDirectories: static func ~file (path: String) -> Int {
+		file := This new(path)
+		result := file createDirectories()
+		file free()
+		result
+	}
+	createParentDirectories: static func ~file (path: String) -> Int {
+		file := This new(path)
+		fileParent := file parent
+		result := fileParent createParentDirectories()
```
Comment (marcusnaslund) [inline source/system/io/File.ooc]: Oops, the error is here, `createDirectories` should be called.

**Comment by marcusnaslund** [comment]
> it could be convenient to have another `Dir` class

Absolutely agree.

**Comment by marcusnaslund** [comment]
As an added bonus, this fixes some memory leaks in the draw tests.

**Pull request merged**